### ISSUE TITLE
element expression breaks on multiline function calls

### ISSUE THREAD

**Issue by diamondburned** [a-h/templ#91]
The following `templ` snippet:

```go
func hello(arg1, arg2 string) string {
	return "hello, " + arg1 + " and " + arg2 + "!"
}

templ x() {
	<p>
		@hello(
			"Alice",
			"Bob"
		)
	</p>
}
```

generates the following Go code:

```go
// Code generated by templ@(devel) DO NOT EDIT.

package main

//lint:file-ignore SA4006 This context is only used if a nested component is present.

import "github.com/a-h/templ"
import "context"
import "io"
import "bytes"

// GoExpression
func hello(arg1, arg2 string) string {
	return "hello, " + arg1 + " and " + arg2 + "!"
}

func x() templ.Component {
	return templ.ComponentFunc(func(ctx context.Context, w io.Writer) (err error) {
		templBuffer, templIsBuffer := w.(*bytes.Buffer)
		if !templIsBuffer {
			templBuffer = templ.GetBuffer()
			defer templ.ReleaseBuffer(templBuffer)
		}
		ctx = templ.InitializeContext(ctx)
		var_1 := templ.GetChildren(ctx)
		if var_1 == nil {
			var_1 = templ.NopComponent
		}
		ctx = templ.ClearChildren(ctx)
		// Element (standard)
		_, err = templBuffer.WriteString("<p>")
		if err != nil {
			return err
		}
		// TemplElement
		err = hello(.Render(ctx, templBuffer)
		if err != nil {
			return err
		}
		// Whitespace (normalised)
		_, err = templBuffer.WriteString(` `)
		if err != nil {
			return err
		}
		// Text
		var_2 := `"Alice",`
		_, err = templBuffer.WriteString(var_2)
		if err != nil {
			return err
		}
		// Whitespace (normalised)
		_, err = templBuffer.WriteString(` `)
		if err != nil {
			return err
		}
		// Text
		var_3 := `"Bob"`
		_, err = templBuffer.WriteString(var_3)
		if err != nil {
			return err
		}
		// Whitespace (normalised)
		_, err = templBuffer.WriteString(` `)
		if err != nil {
			return err
		}
		// Text
		var_4 := `)`
		_, err = templBuffer.WriteString(var_4)
		if err != nil {
			return err
		}
		_, err = templBuffer.WriteString("</p>")
		if err != nil {
			return err
		}
		if !templIsBuffer {
			_, err = io.Copy(w, templBuffer)
		}
		return err
	})
}
```

On L36, `templ` generated this line:

```go
// TemplElement
err = hello(.Render(ctx, templBuffer)
```

It seems that `templ` is tripping on the multiline function call, only reading the first line (`hello(`) and ignoring the rest. This appears to be a bug to me.

---

Use case: I tried doing this:

```go
<formset class="toggles">
	@checkbox("show-comments",
		"Show comments",
		"Show or hide comments on the homepage", data.MeConfig.ShowComments)
	@checkbox(
		"anonymous",
		"Anonymous",
		"Hide your username from other users", data.MeConfig.Anonymous)
</formset>
```

**Comment by diamondburned** [comment]
```html
!{checkbox(
	// ...
)}
```

seems to work.

**Comment by joerdav** [comment]
This is definitely the incorrect behaviour, what version of the templ CLI are you on?
I ask as I cannot reproduce this on the master build.

Just a point to note: the `@` syntax is only used to call other templates (functions that return a `templ.Component`).
https://templ.guide/syntax-and-usage/template-composition

The `{...}` syntax is used for interpolating go expressions, such as strings.
https://templ.guide/syntax-and-usage/elements#attributes-and-elements-can-contain-expressions

**Comment by diamondburned** [comment]
I'm currently on e1ca5e2. I can check the latest commit and report back.

**Comment by diamondburned** [comment]
Awesome! This seems to be fixed as of acea959, so I'm closing this issue. The LSP still seems to be complaining, but that's not important to me.
